agenta-backend/agenta_backend/routers/app_router.py

```diff
@@ -594,8 +594,8 @@ async def create_app_and_variant_from_template(
 
             envvars = {**(payload.env_vars or {})}
             for key in supported_llm_prodviders_keys:
-                if os.environ.get(key):
-                    envvars[key] = os.environ[key]
+                if envvars.get(key):
```

Comment:
I think the right fix is the following
```suggestion
                if not envvars.get(key):
                    envvars[key] = os.environ[key]
```

Reply:
Got it, thanks.